### PR TITLE
VZ-10998 add node pool options to oke capi UI 

### PR DESCRIPTION
**Pull request by knelson7**

### CODE REVIEW

File: lib/shared/addon/components/cluster-driver/driver-okecapi/component.js [hunk 1]
```diff
@@ -146,6 +146,27 @@ export default Component.extend(ClusterDriver, {
...
+    addNodePool() {
+      const newName = `pool-${ this.newRandomizedName() }`
+      let nodePools = get(this, 'nodePools')
+      const np = {
+        name:       newName,
+        replicas:   1,
+        memory:     32,
+        ocpus:      2,
+        volumeSize: 100,
+        shape:      'VM.Standard.E4.Flex',
```
Comment (anders-swanson) [inline lib/shared/addon/components/cluster-driver/driver-okecapi/component.js:158]: needs version, should be the `config.KubernetesVersion` by default

File: lib/shared/addon/components/cluster-driver/driver-okecapi/component.js [hunk 2]
```diff
@@ -535,8 +585,45 @@ export default Component.extend(ClusterDriver, {
...
+
+      return val
+    }
+
+    if (nodePools) {
+      for (let i = 0; i < nodePools.length; i++) {
+        nodePools[i].ocpus = parseIntStrict(nodePools[i].ocpus)
+        nodePools[i].memory = parseIntStrict(nodePools[i].memory)
+        nodePools[i].replicas = parseIntStrict(nodePools[i].replicas)
+        nodePools[i].volumeSize = parseIntStrict(nodePools[i].volumeSize)
```
Comment (anders-swanson) [inline lib/shared/addon/components/cluster-driver/driver-okecapi/component.js:609]: needs version

File: lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs [hunk 3]
```diff
@@ -215,6 +215,121 @@
...
+                {{t 'clusterNew.okecapi.numWorkerNodes.label'}}
+              </label>
+              {{#input-or-display editable=(eq step 2) value=np.replicas}}
+                {{input-integer min=0 max=maxNodeCount value=np.replicas classNames="form-control" placeholder=(t 'clusterNew.ociocne.numWorkerNodes.placeholder')}}
+                <p class="help-block">
+                  {{t 'clusterNew.okecapi.numWorkerNodes.help'}}
+                </p>
+              {{/input-or-display}}
+            </div>
+            <div class="col span-4 mb-0">
```
Comment (anders-swanson) [inline lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs:293]: needs version field

File: translations/en-us.yaml [hunk 4]
```diff
@@ -3771,6 +3771,14 @@ clusterNew:
         label: CNI Type
       apiEndpoint:
         label: Kubernetes API Endpoint Subnet
+    numWorkerNodes:
+      label: Node Count
+      help: The number of nodes in the node pool
+    vcn:
+      quickResources:
+        header: 'New resources include:'
+        k8s: Kubernetes cluster
```
Comment (anders-swanson) [inline translations/en-us.yaml:3782]: unsure these belong, the description of items was meant to be what you'd get for quick create networking (new vcn and all other things).

File: lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs [hunk 5]
```diff
@@ -215,6 +215,134 @@
...
+                </p>
+              {{/input-or-display}}
+            </div>
+          </div>
+          <div class="row">
+            <div class="col span-4 mb-0">
+              <label class="acc-label">
+                {{t 'clusterNew.okecapi.k8sversion.label'}}
+              </label>
+              {{#input-or-display editable=(and (eq step 3) isNew) value=np.memory}}
```
Comment (anders-swanson) [inline lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs]: config.kubernetesVersion

Comment (anders-swanson) [reply lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs]: also, eq step 2

File: lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs [hunk 6]
```diff
@@ -533,6 +663,132 @@
...
+                  </p>
+                {{/input-or-display}}
+              </div>
+            </div>
+            <div class="row">
+              <div class="col span-4 mb-0">
+                <label class="acc-label">
+                  {{t 'clusterNew.okecapi.k8sversion.label'}}
+                </label>
+                {{#input-or-display editable=(and (eq step 3) isNew) value=np.memory}}
```
Comment (anders-swanson) [inline lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs]: value=config.kubernetesVersion

File: lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs [hunk 7]
```diff
@@ -533,6 +663,132 @@
...
+                  {{searchable-select class="form-control"
+                                      content=kubernetesVersionOptions
+                                      value=config.kubernetesVersion
```
Comment (anders-swanson) [inline lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs:709]: value=np.kubernetesVersion

File: lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs [hunk 8]
```diff
@@ -215,6 +215,134 @@
...
+          </div>
+          <div class="row">
+            <div class="col span-4 mb-0">
+              <label class="acc-label">
+                {{t 'clusterNew.okecapi.k8sversion.label'}}
+              </label>
+              {{#input-or-display editable=(and (eq step 3) isNew) value=np.memory}}
+                {{searchable-select class="form-control"
+                                    content=kubernetesVersionOptions
+                                    value=config.kubernetesVersion
```
Comment (anders-swanson) [inline lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs:261]: np.kubernetesVerison

File: lib/shared/addon/components/cluster-driver/driver-ociocne/component.js [hunk 9]
```diff
@@ -689,6 +690,7 @@ export default Component.extend(ClusterDriver, {
         nodePools[i].memory = parseIntStrict(nodePools[i].memory)
         nodePools[i].replicas = parseIntStrict(nodePools[i].replicas)
         nodePools[i].volumeSize = parseIntStrict(nodePools[i].volumeSize)
+        nodePools[i].version = parseIntStrict(nodePools[i].version)
```
Comment (anders-swanson) [inline lib/shared/addon/components/cluster-driver/driver-ociocne/component.js]: this shouldn't be in ocne

File: lib/shared/addon/components/cluster-driver/driver-ociocne/component.js [hunk 10]
```diff
@@ -147,6 +147,7 @@ export default Component.extend(ClusterDriver, {
         ocpus:      2,
         volumeSize: 100,
         shape:      'VM.Standard.E4.Flex',
+        version:    get(this, 'config.kubernetesVersion'),
```
Comment (anders-swanson) [inline lib/shared/addon/components/cluster-driver/driver-ociocne/component.js]: shouldn't be in ocne

File: lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs (hunk 5 above)
Comment (knelson7) [reply lib/shared/addon/components/cluster-driver/driver-okecapi/template.hbs]: shouldn't this also be value=np.kubernetesVersion?

**Review by anders-swanson — APPROVED** [review]

**Pull request merged**